src/Dropdown.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState, useEffect } from "react";
+import cn from "classnames";
+import { Person } from "./types/Person";
+
+type Props = {
+  onSelect: (person: Person) => void;
+  people: Person[];
+};
+
+export const Dropdown: React.FC<Props> = ({ onSelect, people }) => {
+  const [query, setQuery] = useState("");
+  const [debouncedQuery, setDebouncedQuery] = useState("");
+  const [isTyping, setIsTyping] = useState(false);
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      setDebouncedQuery(query);
+      setIsTyping(false);
+    }, 1000);
+
+    return () => {
+      clearTimeout(timerId);
+      setIsTyping(true);
+    };
+  }, [query]);
+
+  const visiblePeople = people.filter((person) => {
+    return person.name.toLowerCase().includes(debouncedQuery.toLowerCase());
+  });
+
+  return (
+    <div className="dropdown is-active">
+      <div className="dropdown-trigger">
+        <input
+          type="text"
+          placeholder="Enter a part of the name"
+          className="input"
+          value={query}
+          onChange={(event) => {
+            setQuery(event.target.value);
+          }}
+        />
+      </div>
+
+      <div className="dropdown-menu" role="menu">
+        <div className="dropdown-content">
+          {debouncedQuery.length > 0 &&
+            !isTyping &&
+            visiblePeople.map((person) => (
+              <button
```

Comment:
add `key`

Reply:
the given data doesn't have id so what i suppose to give as id

Reply:
key is a unique value, it might be a name, an email and so on